flink-shaded-netty-4/src/main/resources/META-INF/NOTICE

```diff
@@ -6,36 +6,37 @@ The Apache Software Foundation (http://www.apache.org/).
 
 This project bundles the following dependencies under the Apache Software License 2.0 (http://www.apache.org/licenses/LICENSE-2.0.txt)
 
-- io.netty:netty-all:4.1.82.Final
-- io.netty:netty-buffer:4.1.82.Final
-- io.netty:netty-codec:4.1.82.Final
-- io.netty:netty-codec-dns:4.1.82.Final
-- io.netty:netty-codec-haproxy:4.1.82.Final
-- io.netty:netty-codec-http:4.1.82.Final
-- io.netty:netty-codec-http2:4.1.82.Final
-- io.netty:netty-codec-memcache:4.1.82.Final
-- io.netty:netty-codec-mqtt:4.1.82.Final
-- io.netty:netty-codec-redis:4.1.82.Final
-- io.netty:netty-codec-smtp:4.1.82.Final
-- io.netty:netty-codec-socks:4.1.82.Final
-- io.netty:netty-codec-stomp:4.1.82.Final
-- io.netty:netty-codec-xml:4.1.82.Final
-- io.netty:netty-common:4.1.82.Final
-- io.netty:netty-handler:4.1.82.Final
-- io.netty:netty-handler-proxy:4.1.82.Final
-- io.netty:netty-resolver:4.1.82.Final
-- io.netty:netty-resolver-dns:4.1.82.Final
-- io.netty:netty-resolver-dns-classes-macos:4.1.82.Final
-- io.netty:netty-resolver-dns-native-macos:osx-x86_64:4.1.82.Final
-- io.netty:netty-resolver-dns-native-macos:osx-aarch_64:4.1.82.Final
-- io.netty:netty-transport:4.1.82.Final
-- io.netty:netty-transport-classes-epoll:4.1.82.Final
-- io.netty:netty-transport-classes-kqueue:4.1.82.Final
-- io.netty:netty-transport-native-epoll:linux-x86_64:4.1.82.Final
-- io.netty:netty-transport-native-epoll:linux-aarch_64:4.1.82.Final
-- io.netty:netty-transport-native-kqueue:osx-x86_64:4.1.82.Final
-- io.netty:netty-transport-native-kqueue:osx-aarch_64:4.1.82.Final
-- io.netty:netty-transport-native-unix-common:4.1.82.Final
-- io.netty:netty-transport-rxtx:4.1.82.Final
-- io.netty:netty-transport-sctp:4.1.82.Final
-- io.netty:netty-transport-udt:4.1.82.Final
\ No newline at end of file
+- io.netty:netty-all:4.1.90.Final
+- io.netty:netty-buffer:4.1.90.Final
+- io.netty:netty-codec-dns:4.1.90.Final
```

Comment:
Have you tested Flink against this netty version?

Reply:
Is there a reasonable set of tests we could run for such a `flink-shaded`. Or would it be just running the entire test suite with the updated `flink-shaded` depdenencies. I'm just asking because it might be worth it to add this to the README.

Reply:
> Have you tested Flink against this netty version?

I can't because I can't run the all the tests in Flink on a Mac M1

Reply:
I did a CI run which does show issues, you can see it here: https://dev.azure.com/martijn0323/Flink/_build/results?buildId=4046&view=results

The Python one looks unrelated, but there's definitely a Netty one

```
Mar 29 14:38:38 [ERROR]   NettyEpollITCase.testNettyEpoll:50->trySetUpCluster:83 » UnsatisfiedLink faile...
```

Reply:
I guess we should also update [<curator.version/> in Flink's parent pom](https://github.com/apache/flink/blob/master/pom.xml#L143) to keep it consistent. You would make the timeout value in [ZooKeeperTestUtils:57](https://github.com/apache/flink/blob/master/flink-runtime/src/test/java/org/apache/flink/runtime/testutils/ZooKeeperTestUtils.java#L57) as a workaround for [CURATOR-663](https://issues.apache.org/jira/browse/CURATOR-663).

On that note: Why do we keep a separate curator dependency for tests? Is it because we don't necessarily need zookeeper in certain modules and want to keep it more light-weight by relying on curator's `TestingZooKeeperServer` implementation?

Reply:
> I guess we should also update [<curator.version/> in Flink's parent pom](https://github.com/apache/flink/blob/master/pom.xml#L143) to keep it consistent.

Yes, but I believe we first have to release this newer version of flink-shaded before we can open the PR towards Flink. 

> On that note: Why do we keep a separate curator dependency for tests? Is it because we don't necessarily need zookeeper in certain modules and want to keep it more light-weight by relying on curator's `TestingZooKeeperServer` implementation?

I defer to @zentol for that

Reply:
{quote}
The Python failure is caused by Pulsar, which uses flink-shaded but that still points to guava30 instead of guava31. I don't think that's a blocker for this PR, but a necessary update that needs to happen in Pulsar and then in PyFlink to point to the new Pulsar artifact.
{quote}
Some but not all errors are caused by a NoClassDefFoundError due to the guava dependency (e.g. [that one](
https://dev.azure.com/martijn0323/Flink/_build/results?buildId=4048&view=logs&j=fba17979-6d2e-591d-72f1-97cf42797c11&t=727942b6-6137-54f7-1ef9-e66e706ea068&l=25602)). Do we know why these classes are missing? :thinking:

Reply:
isn't the reason that they try to import `guava30` like here 

https://github.com/apache/flink-connector-pulsar/blob/343acebd3ec00410747a9c2af3af2dccc87b5f93/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/topic/TopicNameUtils.java#L23-L24

Reply:
> The Python failure is caused by Pulsar, which uses flink-shaded but that still points to guava30 instead of guava31. I don't think that's a blocker for this PR, but a necessary update that needs to happen in Pulsar and then in PyFlink to point to the new Pulsar artifact.

The externalized connectors must not rely on flink-shaded. This will just not be possible if we want them to work against different Flink versions.